CMakeLists.txt

```diff
@@ -163,8 +163,8 @@ else (APPLE)
       set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
       set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
    else (MINGW)
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -fPIC -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG   "-std=gnu++11 -fno-inline -fPIC -g")
+      set(CMAKE_CXX_FLAGS_RELEASE "-std=gnu++11 -fno-inline -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
```

Comment:
Are you 100% sure that you are not using a all.h.gch generated during a debug build to compile a release version? 
Also maybe check the compilation of the precompiled header with make VERBOSE=1 and make sure that the compilation flags are the same for this compilation and for the compilation of any file with this warnings. In particular the -O flag.